test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -195,6 +195,7 @@ public void Dispose()
             _listener?.Stop();
             _cancellationTokenSource.Cancel();
             _udpClient?.Close();
+            _listenerTask.Wait();
```

Comment:
Will this Wait call throw an exception?

Reply:
Not in normal usage, no. In normal usage, the `GetContextAsync()` call throws when we call `_listener.Stop()`, but we swallow it, and exit the loop as `_listener.IsListening`. We _should_ probably have a `?` in there though.